vars/runATH.groovy

```diff
@@ -0,0 +1,221 @@
+ */
+
+def call(Map params = [:]) {
+    def athUrl = params.get('athUrl', 'https://github.com/jenkinsci/acceptance-test-harness.git')
+    def athRevision = params.get('athRevision', 'master')
+    def metadataFile = params.get('metadataFile', 'essentials.yml')
+    def jenkins = params.get('jenkins', 'latest')
+
+    def mirror = "http://mirrors.jenkins.io/"
+    def defaultCategory = "org.jenkinsci.test.acceptance.junit.SmokeTest"
+    def jenkinsURl = jenkins
+    def metadata
+    def athContainerImage
+    def isLocalATH
+    def isVersionNumber
+
+    def athSourcesFolder = "athSources"
+
+    def supportedBrowsers = ["firefox"]
+
+    def skipExecution = false
+
+    def localPluginsStashName = env.RUN_ATH_LOCAL_PLUGINS_STASH_NAME ?: "localPlugins"
+
+    ensureInNode(env, env.RUN_ATH_SOURCES_AND_VALIDATION_NODE ?: "linux", {
+        List<String> env = [
+                "JAVA_HOME=${tool 'jdk8'}",
+                'PATH+JAVA=${JAVA_HOME}/bin',
+                "PATH+MAVEN=${tool 'mvn'}/bin"
+
+        ]
+
+        if (!fileExists(metadataFile)) {
+            echo "Skipping ATH execution because the metadata file does not exist. Current value is ${metadataFile}."
+            skipExecution = true
+            return
+        }
+
+        stage("Getting ATH sources and Jenkins war") {
+            // Start validation
+            metadata = readYaml(file: metadataFile).ath
+            if (metadata == null) {
+                error "The provided metadata file seems invalid as it does not contain a valid ath section"
+            }
+            if (metadata == 'default') {
+                echo "Using default configuration for ATH"
+                metadata = [:]
+            } else if (metadata.browsers == null) {
+                echo "The provided metadata file does not include the browsers property, using firefox as default"
+            }
+            // Allow to override athUrl and athRevision from metadata file
+            athUrl = metadata.athUrl ?: athUrl
+            isLocalATH = athUrl.startsWith("file://")
+            athRevision = metadata.athRevision ?: athRevision
+
+            // Allow override of jenkins version from metadata file
+            jenkins = metadata.jenkins ?: jenkins
+            isVersionNumber = (jenkins =~ /^\d+([.]\d+)*$/).matches()
+
+            if (!isLocalATH) {
+                echo 'Checking connectivity to ATH sources…'
+                sh "git ls-remote --exit-code -h ${athUrl}"
+            }
+            if (jenkins == "latest") {
+                jenkinsURl = mirror + "war/latest/jenkins.war"
+            } else if (jenkins == "latest-rc") {
+                jenkinsURl = mirror + "/war-rc/latest/jenkins.war"
+            } else if (jenkins == "lts") {
+                jenkinsURl = mirror + "war-stable/latest/jenkins.war"
+            } else if (jenkins == "lts-rc") {
+                jenkinsURl = mirror + "war-stable-rc/latest/jenkins.war"
+            }
```

Comment:
should there be an else { error 'wat' } to avoid surprises?

Reply:
If none of this if's match I am assuming jenkinsUrl is a URI, not failing in that case because I want to support URI's like links to files generated by other jobs, there is a validation to check if the file is reachable (valid URL) on line 80